manifests/server/config.pp

```diff
@@ -9,21 +9,42 @@
     section => 'master',
   }
 
-  if $puppet::server::environmentpath {
-    $env_ensure = 'present'
-    $mod_ensure = 'absent'
+  if $puppet::server::directoryenvs == true or $puppet::server::directoryenvs == 'true' {
```

Comment:
Instead of this, how about validate_bool in server.pp, and here check only `if $puppet::server::directoryenvs`?